src/mscorlib/shared/System/IO/Path.Windows.cs

```diff
@@ -223,5 +223,42 @@ public static ReadOnlySpan<char> GetPathRoot(ReadOnlySpan<char> path)
 
         /// <summary>Gets whether the system is case-sensitive.</summary>
         internal static bool IsCaseSensitive { get { return false; } }
+
+        internal static ReadOnlySpan<char> GetVolumeName(ReadOnlySpan<char> path)
+        {
+            if (!IsPathFullyQualified(path))
+                return default;
+
+            // 3 cases: UNC ("\\server\share"), Device ("\\?\C:\"), or Dos ("C:\")
+            ReadOnlySpan<char> root = GetPathRoot(path);
+            if (!PathInternal.IsDevice(path))
```

Comment:
You want `server\share` out of any type of UNC, in device syntax or not. This would return `\\server\share`.

Reply:
You could, in theory, pass back `\\server\share` for all UNC types. Doing so, however, would require allocating in the device unc format. As directory separators aren't valid inside a name, it makes trimming it on both ends workable.